node/core/dispute-coordinator/src/real/mod.rs

```diff
@@ -648,21 +861,36 @@ async fn handle_import_statements(
 
 			return Ok(ImportStatementsResult::InvalidImport)
 		},
-		Some(info) => info.validators.clone(),
+		Some(info) => info,
 	};
+	let validators = session_info.validators.clone();
 
 	let n_validators = validators.len();
 
 	let supermajority_threshold = polkadot_primitives::v1::supermajority_threshold(n_validators);
 
-	let mut votes = overlay_db
+	// In case we are not provided with a candidate receipt
+	// we operate under the assumption, that a previous vote
+	// which included a `CandidateReceipt` was seen.
+	// This holds since every block is preceeded by the `Backing`-phase.
+	let mut votes = match overlay_db
 		.load_candidate_votes(session, &candidate_hash)?
 		.map(CandidateVotes::from)
-		.unwrap_or_else(|| CandidateVotes {
-			candidate_receipt: candidate_receipt.clone(),
-			valid: Vec::new(),
-			invalid: Vec::new(),
-		});
+	{
+		Some(votes) => votes,
+		None =>
+			if let MaybeCandidateReceipt::Provides(candidate_receipt) = candidate_receipt {
+				CandidateVotes { candidate_receipt, valid: Vec::new(), invalid: Vec::new() }
+			} else {
+				tracing::warn!(
+					target: LOG_TARGET,
+					session,
+					"Missing info for session which has an active dispute",
+				);
+				return Ok(ImportStatementsResult::InvalidImport)
```

Comment:
I don't really like that this discards the votes, but as mentioned above it's very difficult to reach this code path. An ideal solution would be for `CandidateVotes` to tolerate not having the candidate receipt, but that we'd only participate once we have the candidate receipt.

Reply:
This is not a blocker but I would like a follow-up issue for this at the very least.

Reply:
https://github.com/paritytech/polkadot/issues/4011